### PR TITLE
Upgrade to php 8, Laravel 8

### PR DESCRIPTION
**Pull request by jclarkkoala**
## What
Update the Pouch package to work with PHP 7.4/8.0 and Laravel 8+.

Fixed broken functionality related to many-to-many relationships and sorting of nested nested relationships

### CODE REVIEW

File: .php_cs.dist.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * Suggested code style guidelines
+ *
+ * - https://github.com/FriendsOfPHP/PHP-CS-Fixer
+ * - https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
+ */
+
+$finder = PhpCsFixer\Finder::create()
```
Comment (walterbm) [inline .php_cs.dist.php:10]: Than you for adding linting!

File: .php_cs.dist.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    ->exclude('templates')
+    ->exclude('bootstrap')
+    ->exclude('public')
+    ->exclude('private')
+    ->exclude('resources')
+    ->exclude('vendor')
+    ->exclude('database/seeds/configs')
```
Comment (walterbm) [inline .php_cs.dist.php]: Can we remove these explicit exclusions? Seems like they're only relevant in an actual Laravel app  
```suggestion
```

File: src/Filter.php [hunk 3]
```diff
@@ -13,544 +14,551 @@
...
+            } elseif ($filter === 'true' || $filter === 'false') {
+                // Is a boolean filter, coerce to boolean.
+                $filter = ($filter === 'true');
+
+                // Querying a dot nested relation
+                if (is_array($nested_relations)) {
+                    $query->whereHas(
+                        $relation,
+                        function ($query) use ($filter, $column) {
+                            $where = ('where' . $column);
```
Comment (walterbm) [inline src/Filter.php]: Looks like this used to have a call to `camel_case` — was that unnecessary?

**Review by walterbm — APPROVED** [review]
This looks excellent! Liniting everything is a huge improvement and thank you for fixing the tests!

File: src/Filter.php (hunk 3 above)
Comment (jclarkkoala) [reply src/Filter.php]: Laravel global namespace helper functions like `camel_case` and `array_reverse` were deprecated back in Laravel 6. They are now accessible through support static classes `Str` and `Arr`.

Comment (walterbm) [reply src/Filter.php]: @jclarkkoala Makes sense — so should this have a `Str::camel` to replace the previous `camel_case`?

Comment (jclarkkoala) [reply src/Filter.php]: Resolved

**Pull request merged**